### ISSUE TITLE
[DependencyInjection] Add possibility to avoid creating of `preload.php`

### ISSUE THREAD

**Issue by alexander-schranz** [symfony/symfony#38334]
**Symfony version(s) affected**: `4.4.14` / `5.1.6`

**Description**  

With `4.4.14` / `5.1.6` symfony generate a new file in the `src` called `.preload.php` it was introduced by #38063. To allow a general configuration for hosters to preload a file. As I think its a great feature, but sadly in some setups preload is not possible, this is for example currently the state for sulu. That there still a `.preload.php` is generated currently its missleading devs thinking we are now supporting it, so it would be great to have an option to disable this generation.

**How to reproduce**  

```bash
bin/console cache:clear # generate `src/preload.php`
```

**Possible Solution**  

Add a container parameter to disable this behaviour, or make the path to the `preload` file configurable. Instead of a container parameter a `$_SERVER` or `$_ENV` var could be used.

**Comment by niconoe-** [comment]
I totally agree that this file is a gem, but its activation and its path must both be configurable.

**Comment by iquito** [comment]
This is also bad in development (so an option to turn it off should exist) as it leads static analyzers like Psalm and PHPStan to load and analyze all the preload files. My Psalm execution went from 3 seconds to 95 seconds with 2900 errors because of the .preload.php file.

While preloading and a preloading file is great, Symfony writing into /src should only happen with an explicit opt-in.

**Comment by wouterj** [comment]
Doesn't this also mean the user running `cache:clear` must have write access to the `src/` directory? If so, this would be a counter move compared to the recent cache/build dir split PR (https://github.com/symfony/symfony/pull/36515).

Is there a particular reason to put this file in `src/`, rather than `var/`? With my little knowledge on this topic, I can see advantages of placing it in `var/preload.php`: it's a predictable filename (fixing the original issue); doesn't require using a small hack to avoid it from being loaded by PSR-4; generates file in the directory a user is expecting Symfony to generate files in; and isn't a hidden file (so easier to find for new Symfony devs).

**Comment by iquito** [comment]
The `var/` directory does not exist from the Symfony / kernel perspective (see [the response from Nicolas Grekas](https://github.com/symfony/symfony/pull/38063#issuecomment-700031761) about this), it only knows the project directory, cache directory, and the directory of the kernel.

Yet I think none of those options are good - writing into the directory of the kernel is unexpected and runs counter to read-only source directories, and the cache directory caused the problems mentioned in the pull request. From my perspective the location of the preload file should be explicitely defined in configuration (then it could easily be set to `/var/preload.php`) and only generated when configured, because the developer also needs to add it to php.ini, so someone wanting to use preloading needs to do some configuration and have some knowledge of the feature. The generated Symfony file is never enough anyway.

**Comment by nicolas-grekas** [comment]
Agreed, see #38343 and https://github.com/symfony/recipes/pull/825

**Comment by alexander-schranz** [comment]
@nicolas-grekas Thank you! Find the recipe a great solution!

**Comment by ovrflo** [comment]
I'm throwing my 2c here too.

Yes, the path for the preload is not great and may require some explaining to the ops guys, but as long as it's fixed it's not that bad. Trouble starts when symfony renames the files in cache (which it did in the past quite a few times)

Personally, I've been running with preload for the last 6-7 months without issues.

Regarding the `src/.preload.php`, my production environment will not complain using it but I will complain having to see it there daily in the IDE 😄 

TL;DR; the recipe including config/preload.php seems like a really good compromise (for 4.4/5.x, at least).

Thanks, nicolas 👍
